### PR TITLE
test: HIP-904 Create HAPI tests for a hollow account on an alias on which we have a deleted account

### PR DESCRIPTION
**Pull request by zhpetkov**
**Description**:
Add hollow account tests on an alias on which we have a deleted (hollow) account via transferring HBAR and 1 Fungible Token.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 1]
```diff
@@ -2367,4 +2373,170 @@ final Stream<DynamicTest> createHollowAccountWithNftTransferAndCompleteIt() {
...
+                        final var newAccountID = hapiGetTxnRecord
+                                .getFirstNonStakingChildRecord()
+                                .getReceipt()
+                                .getAccountID();
+                        spec.registry().saveAccountId(hollowAccountKey, newAccountID);
+                    }
+                    // Verify maxAutomaticAssociations is set to 0
+                    var getInfo = getAccountInfo(hollowAccountKey)
+                            .hasAlreadyUsedAutomaticAssociations(0)
+                            .has(accountWith().key(hollowAccountKey))
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2423]: Hollow account will have no key, please correct the assertion.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 2]
```diff
@@ -2367,4 +2373,170 @@ final Stream<DynamicTest> createHollowAccountWithNftTransferAndCompleteIt() {
...
+                            .hasMaxAutomaticAssociations(0)
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2424]: If we wanna have hollow account with maxAutoAssociation = 0, initially we need to be working with maxAutoAssociations flag == false. Then after the first delete we need to find out if we can switch the flag to true before executing the next hollow account creation. After it the hollow account created will have maxAutoAssociations = 1.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 3]
```diff
@@ -2367,4 +2373,170 @@ final Stream<DynamicTest> createHollowAccountWithNftTransferAndCompleteIt() {
...
+
+                    // Sends HBAR from hollow account
+                    var hollowAccountTransferHBAR = cryptoTransfer(
+                                    movingHbar(ONE_HBAR).between(hollowAccountKey, TREASURY))
+                            .payingWith(hollowAccountKey)
+                            .signedBy(hollowAccountKey, TREASURY)
+                            .sigMapPrefixes(uniqueWithFullPrefixesFor(hollowAccountKey))
+                            .via(transferHBARSToHollowAccountTxn);
+
+                    allRunFor(spec, hollowCreate2, getInfo2, hollowAccountTransferHBAR);
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2455]: Here we are missing one operation that verifies if the hollow accounts is completed and has the correct maxAutoAssociations

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 4]
```diff
@@ -2367,4 +2373,170 @@ final Stream<DynamicTest> createHollowAccountWithNftTransferAndCompleteIt() {
...
+                                    .getKey(hollowAccountKey)
+                                    .getECDSASecp256K1()
+                                    .toByteArray();
+                            final var evmAddressBytes = ByteString.copyFrom(recoverAddressFromPubKey(ecdsaKey));
+                            hollowAccountAlias.set(evmAddressBytes);
+                            fungibleTokenId.set(registry.getTokenID(FUNGIBLE_TOKEN));
+                        }))
+                .when(withOpContext((spec, opLog) -> {
+                    // Create a hollow account
+                    var hollowCreate = cryptoCreate(hollowAccountKey)
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2492]: Same as above, fix hollow account creation.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 5]
```diff
@@ -2367,4 +2373,170 @@ final Stream<DynamicTest> createHollowAccountWithNftTransferAndCompleteIt() {
...
+                    if (!hapiGetTxnRecord.getChildRecords().isEmpty()) {
+                        final var newAccountID = hapiGetTxnRecord
+                                .getFirstNonStakingChildRecord()
+                                .getReceipt()
+                                .getAccountID();
+                        spec.registry().saveAccountId(hollowAccountKey, newAccountID);
+                    }
+                    // Verify maxAutomaticAssociations is set to 0
+                    var getInfo = getAccountInfo(hollowAccountKey)
+                            .hasAlreadyUsedAutomaticAssociations(0)
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2511]: Same as above, fix assertions

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java (hunk 2 above)
Comment (zhpetkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2424]: Where is the flag for my case? I see a flag only in CryptoUpdateSuite.

Comment (kselveliev) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java]: This is the flag -> overriding("entities.unlimitedAutoAssociationsEnabled", TRUE_VALUE),

Comment (zhpetkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:2425]: Thank you!

**Comment by kselveliev** [comment]
Need to fix merge conflicts

**Review by kselveliev — DISMISSED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by kselveliev — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Comment by kselveliev** [comment]
Merging as all sides approved and all checks pass

**Pull request merged**